i6_models/parts/frontend/vgg_act.py

```diff
@@ -0,0 +1,234 @@
+
+        :param tensor: input tensor of shape [B,T,F]
+        :param sequence_mask: the sequence mask for the tensor
+        :return: torch.Tensor of shape [B,T",F'] and the shape of the sequence mask
+        """
+        assert tensor.shape[-1] == self.cfg.in_features
+        # and add a dim
+        tensor = tensor[:, None, :, :]  # [B,C=1,T,F]
+
+        tensor = self.conv1(tensor)
+        sequence_mask = mask_pool(
+            seq_mask=sequence_mask,
+            kernel_size=get_int_tuple_int(self.conv1.kernel_size, 0),
+            stride=get_int_tuple_int(self.conv1.stride, 0),
+            padding=get_int_tuple_int(self.conv1.padding, 0),
+        )
+
+        tensor = self.conv2(tensor)
+        sequence_mask = mask_pool(
+            sequence_mask,
+            get_int_tuple_int(self.conv2.kernel_size, 0),
+            get_int_tuple_int(self.conv2.stride, 0),
+            get_int_tuple_int(self.conv2.padding, 0),
+        )
+
+        tensor = self.activation(tensor)
+        tensor = self.pool1(tensor)  # [B,C,T',F']
+        sequence_mask = mask_pool(
+            sequence_mask,
+            get_int_tuple_int(self.pool1.kernel_size, 0),
+            get_int_tuple_int(self.pool1.stride, 0),
+            get_int_tuple_int(self.pool1.padding, 0),
+        )
+
+        tensor = self.conv3(tensor)
+        sequence_mask = mask_pool(
+            sequence_mask,
+            get_int_tuple_int(self.conv3.kernel_size, 0),
+            get_int_tuple_int(self.conv3.stride, 0),
+            get_int_tuple_int(self.conv3.padding, 0),
+        )
+
+        tensor = self.conv4(tensor)
+        sequence_mask = mask_pool(
+            sequence_mask,
+            get_int_tuple_int(self.conv4.kernel_size, 0),
+            get_int_tuple_int(self.conv4.stride, 0),
+            get_int_tuple_int(self.conv4.padding, 0),
+        )
+
+        tensor = self.activation(tensor)
+        tensor = self.pool2(tensor)  # [B,C,T",F"]
+        sequence_mask = mask_pool(
+            sequence_mask,
+            get_int_tuple_int(self.pool2.kernel_size, 0),
+            get_int_tuple_int(self.pool2.stride, 0),
+            get_int_tuple_int(self.pool2.padding, 0),
+        )
+
+        tensor = torch.transpose(tensor, 1, 2)  # transpose to [B,T",C,F"]
+        tensor = torch.flatten(tensor, start_dim=2, end_dim=-1)  # [B,T",C*F"]
+
+        tensor = self.linear(tensor)
+
+        return tensor, sequence_mask
+
+    def _calculate_dim(self) -> int:
+        out_dim = calculate_output_dim(
+            in_dim=self.cfg.in_features,
+            filter_size=get_int_tuple_int(self.pool1.kernel_size, 1),
+            stride=get_int_tuple_int(self.pool1.stride, 1),
+        )
+        out_dim = calculate_output_dim(
+            in_dim=out_dim,
+            filter_size=get_int_tuple_int(self.pool2.kernel_size, 1),
+            stride=get_int_tuple_int(self.pool2.stride, 1),
+        )
```

Comment:
Here you again have the implicit assumption that there is no striding in the convolutions. My same arguments as earlier would apply, that it is better to avoid as much implicit assumptions as possible.